shared/fs/common/folder-view-filter.js

```diff
@@ -6,31 +6,37 @@ import * as Kb from '../../common-adapters'
 import * as FsGen from '../../actions/fs-gen'
 import * as Styles from '../../styles'
 import * as React from 'react'
+import {debounce} from 'lodash-es'
 
 type Props = {|
-  filter: string,
   onUpdate: string => void,
   path: Types.Path,
   style?: ?Styles.StylesCrossPlatform,
 |}
 
 type State = {|
   editing: boolean,
+  filter: string,
 |}
 
 const KeyHandler = (Platforms.isMobile ? c => c : require('../../util/key-handler.desktop').default)(
   () => null
 )
 
 class FolderViewFilter extends React.PureComponent<Props, State> {
-  state = {editing: false}
+  state = {
+    editing: false,
+    filter: '',
+  }
+
+  _update = text => {
+    this.setState({filter: text})
+    this.props.onUpdate(text)
+  }
 
   _input = React.createRef()
 
-  _onBlur = () => {
-    this.setState({editing: false})
-    this.props.onUpdate('')
```

Comment:
This was the bug. We were sending the update in _onBlur, which happens when click on a row happens. So just move this into the Escape handler and only clear the filter if user presses escape when in the filter.